presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -433,12 +434,13 @@ public PlanOptimizers(
                         .build()));
 
         if (!forceSingleNode) {
+            builder.add(new ReplicateSemiJoinInDelete()); // Must run before AddExchanges and before DetermineSemiJoinDistributionType
```

Comment:
> and before DetermineSemiJoinDistributionType

Not quite true. `ReplicateSemiJoinInDelete` would work the same if run after `DetermineSemiJoinDistributionType`